### PR TITLE
Extract endpoint functionality to a separate module

### PR DESCRIPTION
**Pull request by javidq**
`Auth::_endpoint` and its usage is extracted to the dedicated module `host.py`, while keeping environment switch in Auth class to avoid breaking sdk tests. The change is motivated by simplification of http related functionality to improve testability of the design. The functionality and consumer API hasn't been changed in the refactoring process.

Items:
* [x] Ran test & live-tests
* [x] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

File: up42/auth.py [hunk 1]
```diff
@@ -197,15 +193,23 @@ def _get_token_project_based(self):
...
         self.token = token_response["data"]["accessToken"]
 
+    @property
+    def env(self):
+        return host.DOMAIN
+
+    @env.setter
+    def env(self, value: str):
+        host.DOMAIN = value
+
```
Comment (javidq) [inline up42/auth.py:209]: This change is not needed for end users, but it's needed for sdk tests

Comment (javidq) [reply up42/auth.py:209]: Will be dropped when sdk tests are adapted

**Comment by seedlit** [comment]
@javidq since we are hardcoding the domain in `host.py`, would it still work for staging environment?

**Comment by javidq** [comment]
> @javidq since we are hardcoding the domain in `host.py`, would it still work for staging environment?

Yes, since the keep the same interface via env parameter in auth construction. After this PR is merged I will create a PR on SDK tests to manipulate host.env directly and drop it from auth completely

**Review by seedlit — APPROVED** [review]
looked over the surface.
LGTM!

**Pull request merged**